model/registry/hardware.yaml

```diff
@@ -0,0 +1,113 @@
+groups:
+  - id: registry.hardware
+    prefix: hw
+    type: attribute_group
+    brief: >
+      Attributes for hardware.
+    attributes:
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          An identifier for the hardware component, unique within the monitored host
+        examples: ['win32battery_battery_testsysa33_1']
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          An easily-recognizable name for the hardware component
+        examples: ['eth0']
+      - id: parent
+        type: string
+        stability: experimental
+        brief: >
+          Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller)
+        examples: ['dellStorage_perc_0']
+      - id: type
+        type:
+          allow_custom_values: true
+          members:
+            - id: battery.yaml
+              value: 'battery'
+              brief: "Battery"
+              stability: experimental
+            - id: cpu
+              value: 'cpu'
+              brief: 'CPU'
+              stability: experimental
+            - id: disk_controller
+              value: 'disk_controller'
+              brief: 'Disk controller'
+              stability: experimental
+            - id: enclosure
+              value: 'enclosure'
+              brief: 'Enclosure'
+              stability: experimental
+            - id: fan
+              value: 'fan'
+              brief: 'Fan'
+              stability: experimental
+            - id: gpu
+              value: 'gpu'
+              brief: 'GPU'
+              stability: experimental
+            - id: logical_disk
+              value: 'logical_disk'
+              brief: 'Logical disk'
+              stability: experimental
+            - id: memory
+              value: 'memory'
+              brief: 'Memory'
+              stability: experimental
+            - id: network
+              value: 'network'
+              brief: 'Network'
+              stability: experimental
+            - id: physical_disk
+              value: 'physical_disk'
+              brief: 'Physical disk'
+              stability: experimental
+            - id: power_supply
+              value: 'power_supply'
+              brief: 'Power supply'
+              stability: experimental
+            - id: tape_drive
+              value: 'tape_drive'
+              brief: 'Tape drive'
+              stability: experimental
+            - id: temperature
```

Comment:
hw.type = temperature? What it be used for? cooling system? temperature sensor?

Reply:
it is defined as temperature sensor in the hardware-metrics. The same goes for voltage

Reply:
@lmolkova It's a matter of opinion: `temperature` is not confusing to me. 

Appending `_sensor` doesn't provide additional meaning, as most metrics in hardware are reported by "sensors" anyway: power status, memory presence, temperature, fan speed, etc. Why add `_sensor` for temperatures and voltages, but not for fan speed or power supplies? Let's keep this simple and consistent.

Reply:
`hw.type = fan`  makes sense, fan is a hardware component, same with `power_supply`.
`temperature` is not a component.

Reply:
Okay @lmolkova now I understand why it sounds confusing. Actually, `hw.type` is not to be read as "hardware type", but as the "type" attribute used by various metrics (notably `hw.status`), which happens to be in the `hw.` namespace.

`hw.status` can be used for disks, network cards, memory modules, etc. and temperatures, voltages, etc. `hw.status{hw.type="temperature"}` will report whether temperature is okay or not, and not whether the temperature sensor is operating properly. `hw.status{hw.type="temperature_sensor"}` would actually be very confusing to me, as I'd interpret it as whether the sensor is working fine or not.

I'm not sure however if we can clarify this in the description of the attribute. I still hope my explanation above makes it clearer! 😅

Reply:
I see, thanks for the explanation!

I agree that `temperature_sensor` would not make sense either then.

I guess that's another big thing to revisit in the https://github.com/open-telemetry/semantic-conventions/issues/1071 and should not block this PR.

So for the future
- `hw.type` name could be more precise. But also wouldn't it be clearer to report temperature as a separate metric (attribute on spans) and let someone decided if it's good/bad at a query time?  `hw.type=temperature, hw.status=failed` does not quite make sense either
- we might need to break this enum down into a component type (cpu|fan|etc) & component state|indicator|etc?. E.g. if I want to report CPU temperature, how do I do it today? `name="AMD Rizen 42",type=temperature,status=ok` ?

For this PR:

can we clarify what's `hw.type` is in the `hw.type` note?  E.g.

> Describes the category of hardware component for which `hw.state` is reported. For example, `hw.type=temperature` along with `hw.state=degraded` describes that the temperature of the hardware component has degraded.